org-code-javabuilder/lib/src/main/java/dev/javabuilder/WebSocketServer.java

```diff
@@ -38,21 +38,26 @@
    */
   @OnOpen
   public void onOpen(Session session) {
-    Map<String, List<String>> params = session.getRequestParameterMap();
-    String channelId = params.get("channelId").get(0);
+    // Decode the authorization token
+    String token = session.getRequestParameterMap().get("Authorization").get(0);
+    Base64.Decoder decoder = Base64.getDecoder();
+    String payload = new String(decoder.decode(token.split("\\.")[1]));
+    JSONObject queryInput = new JSONObject(payload);
+
+    final String levelId = queryInput.getString("level_id");
+    final String channelId = queryInput.getString("channel_id");
+    final String dashboardHostname = "http://" + queryInput.get("iss") + ":3000";
+    final JSONObject options = new JSONObject(queryInput.get("options"));
     boolean useNeighborhood = false;
-    if (params.containsKey("useNeighborhood")) {
-      useNeighborhood = Boolean.parseBoolean(params.get("useNeighborhood").get(0));
-    }
-    String levelId = "";
-    if (params.containsKey("levelId")) {
-      levelId = params.get("levelId").get(0);
+    if (options.has("useNeighborhood")) {
+      String useNeighborhoodStr = options.getString("useNeighborhood");
+      useNeighborhood = Boolean.parseBoolean(useNeighborhoodStr);
     }
+
```

Comment:
This section was updated to match the LambdaRequestHandler, the entry point for production Javabuilder
https://github.com/code-dot-org/javabuilder/blob/5cebd2f38f51437f177cd2e76a875165d8a4ddc5/org-code-javabuilder/lib/src/main/java/org/code/javabuilder/LambdaRequestHandler.java#L43-L54

Reply:
Is it possible to share this code with `LambdaRequestHandler` as it has a similar function? Seems fine if not, since it's not too much, but just asking to help prevent any potential divergences here between local/prod.

Reply:
Yes, I think the eventual goal would be to pass the options object into the various classes/methods that need it rather than querying its fields here. This would provide two benefits: it would further align the WebSocket and Lambda deployments and it would remove the need for testing on an AWS deployed javabuilder each time a new field is added to options.

We generally [prefer small PRs](https://docs.google.com/document/d/1-x-GcC2xWpUymU6YBWHzxmTAlASWQPMfAf8ASBqWlgA/edit). This seemed like a good atomic improvement that can be built upon in the future and a good pausing point in this work.

Reply:
Got it, makes sense!